### PR TITLE
add revrec to addons and items

### PR DESCRIPTION
**Pull request by wsethbrown**
Add revrec to plan add ons and items

### CODE REVIEW

**Review by faiberrec — COMMENTED** [review]
There are missing tests for Item create/update. The same goes for add ons.

File: spec/recurly/add_on_spec.rb [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+      add_on_code:                'pink_sweaters',
+      name:                       'Pink Sweaters',
+      revenue_schedule_type:      'evenly',
+      unit_amount_in_cents:       200,
+      add_on_type:                'usage',
+      optional:                   false,
+      usage_type:                 'price',
+      liability_gl_account_id:    'uf0jwj5zhclg',
+      revenue_gl_account_id:      'uf0jwincednb',
+      performance_obligation_id:   1,
```
Comment (Smolations) [inline spec/recurly/add_on_spec.rb]: This should technically be a string.

File: spec/recurly/item_spec.rb [hunk 2]
```diff
@@ -10,7 +10,10 @@
       accounting_code: '1234',
       revenue_schedule_type: 'evenly',
       avalara_transaction_type: 600,
-      avalara_service_type: 3
+      avalara_service_type: 3,
+      liability_gl_account_id: 'uf0jwj5zhclg',
+      revenue_gl_account_id: 'uf0jwincednb',
+      performance_obligation_id: 1
```
Comment (Smolations) [inline spec/recurly/item_spec.rb]: This should technically be a string.

File: spec/recurly/add_on_spec.rb [hunk 3]
```diff
@@ -85,5 +122,48 @@
...
+        optional: false,
+        usage_type: 'price',
+        liability_gl_account_id: 'uf0jwj5zhclg',
+        revenue_gl_account_id: 'uf0jwincednb',
+        performance_obligation_id: '1'
+      )
+      add_on.must_be_instance_of AddOn
+      add_on.name.must_equal 'Pink Sweaters'
+    end
+  end
```
Comment (faiberrec) [inline spec/recurly/add_on_spec.rb]: We are missing a call to `.create` in this example.

File: spec/recurly/add_on_spec.rb [hunk 4]
```diff
@@ -85,5 +122,48 @@
...
+
+  describe "update" do
+    it "sends changed attributes to the server" do
+      plan = Plan.find 'gold'
+      add_on = plan.add_ons.find 'marketing_email'
+      stub_request(:put, "https://api.recurly.com/v2/plans/gold/add_ons/marketing_email").
+      with(:body => "<add_on><name>Updated Emails</name><revenue_gl_account_id>aksdfu48</revenue_gl_account_id></add_on>",
+        :headers => Recurly::API.headers).to_return(:status => 200, :body => "", :headers => {})
+      add_on.update_attributes({ name: 'Updated Emails', revenue_gl_account_id: 'aksdfu48' })
```
Comment (faiberrec) [inline spec/recurly/add_on_spec.rb]: would be nice to add the other RevRec attributes we can update `liability_gl_account_id` and `performance_obligation_id`.

File: spec/recurly/item_spec.rb [hunk 5]
```diff
@@ -57,9 +92,9 @@
       it "sends changed attributes to the server" do
         stub_api_request :get, 'items/plastic_gloves', 'items/show-200'
         stub_request(:put, "https://api.recurly.com/v2/items/plastic_gloves").
-        with(:body => "<item><name>Sleek Plastic Gloves</name></item>",
+        with(:body => "<item><name>Sleek Plastic Gloves</name><revenue_gl_account_id>aksdfu48</revenue_gl_account_id></item>",
           :headers => Recurly::API.headers).to_return(:status => 200, :body => "", :headers => {})
-        item.update_attributes({ name: 'Sleek Plastic Gloves' })
+        item.update_attributes({ name: 'Sleek Plastic Gloves', revenue_gl_account_id: 'aksdfu48' })
```
Comment (faiberrec) [inline spec/recurly/item_spec.rb]: would be nice to add the other RevRec attributes we can update liability_gl_account_id and performance_obligation_id.

File: spec/recurly/add_on_spec.rb (hunk 4 above)
Comment (wsethbrown) [reply spec/recurly/add_on_spec.rb]: Done!

File: spec/recurly/item_spec.rb (hunk 5 above)
Comment (wsethbrown) [reply spec/recurly/item_spec.rb]: Done!

File: spec/recurly/add_on_spec.rb (hunk 3 above)
Comment (wsethbrown) [reply spec/recurly/add_on_spec.rb]: Updated

**Review by faiberrec — APPROVED** [review]

**Review by faiberrec — APPROVED** [review]

**Comment by 8eth** [comment]
DevQA+1

**Pull request merged**